drivers: regulator: sf32lb: Add sf32lb peri ldo driver

HalfSweet opened the pull request:
Add sf32lb peri ldo driver
- LDO1V8
- LDO2_3V3
- LDO3_3V3

seov-nordic commented on drivers/regulator/regulator_sf32lb.c:
```diff
@@ -0,0 +1,144 @@
+	}
+
+	/* Return min voltage from common config (fixed regulator) */
+	*volt_uv = config->common.min_uv;
```
`regulator-min-microvolt` is not a required property, therefore could be unset which results in `INT32_MIN`, see https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/regulator.h#L199-L200

take a look at regulator_fixed, may be worth implementing this in a similar way https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/regulator/regulator_fixed.c#L55-L75

ZhaoxiangJin commented on drivers/regulator/regulator_sf32lb.c:
```diff
@@ -0,0 +1,144 @@
+LOG_MODULE_REGISTER(regulator_sf32lb, CONFIG_REGULATOR_LOG_LEVEL);
+
+/* Number of LDO rails */
+#define SF32LB_PERI_LDO_COUNT 3
```
Why use a macro instead of a DT property?

HalfSweet replied:
done

gmarull commented on dts/bindings/regulator/sifli,sf32lb-peri-ldo.yaml:
```diff
@@ -0,0 +1,65 @@
+  reg:
+    required: true
+    description: |
+      Address of the PERI_LDO register in PMUC.
```
create a child node in PMUC, with offset address

gmarull commented on dts/bindings/regulator/sifli,sf32lb-peri-ldo.yaml:
```diff
@@ -0,0 +1,65 @@
+  - ldo3-3v3: 3.3V LDO3 for peripherals
+
+  Example:
+    peri_ldo: peri-ldo@500ca05c {
```
nodelabel not needed?, use standard naming, like `regulator@...`

HalfSweet replied:
done

gmarull commented on drivers/regulator/Kconfig.sf32lb:
```diff
@@ -0,0 +1,22 @@
+# Copyright (c) 2025 SiFli Technologies(Nanjing) Co., Ltd
+# SPDX-License-Identifier: Apache-2.0
+
+config REGULATOR_SF32LB
```
`REGULATOR_SF32LB_LDO`?

HalfSweet committed:
dts: bindings: sifli: add SF32LB52x PMUC and LDO bindings

Rename PMUC compatible to be SF32LB52x specific.

Add devicetree bindings and dt-bindings
helpers for the SF32LB52x peripheral LDO regulators

Signed-off-by: Haoran Jiang <halfsweet@halfsweet.cn>

HalfSweet committed:
dts: arm: sifli: add SF32LB52x peripheral LDO regulators

Add the PMUC peripheral LDO regulators as a child 'regulators' node.

Signed-off-by: Haoran Jiang <halfsweet@halfsweet.cn>

HalfSweet committed:
drivers: regulator: sf32lb: add SF32LB52x peripheral LDO driver

Add a regulator driver for the SF32LB52x internal peripheral LDOs
controlled by PMUC

Signed-off-by: Haoran Jiang <halfsweet@halfsweet.cn>

gmarull reviewed (commented):
use child node based instantiation (see e.g. npm1300)

HalfSweet commented on dts/bindings/regulator/sifli,sf32lb52x-ldo.yaml:
```diff
@@ -0,0 +1,63 @@
+        #size-cells = <0>;
+
+        ldo18: regulator@0 {
+          reg = <SF32LB52X_LDO18>;
```
This is because we need to know which LDO to enable

HalfSweet commented on drivers/regulator/regulator_sf32lb.c:
```diff
@@ -0,0 +1,149 @@
+
+LOG_MODULE_REGISTER(regulator_sf32lb, CONFIG_REGULATOR_LOG_LEVEL);
+
+#define DT_DRV_COMPAT sifli_sf32lb52x_ldo
```
done
